lib/pos_blockchain.go

```diff
@@ -374,17 +366,51 @@ func (bc *Blockchain) getLineageFromCommittedTip(desoBlock *MsgDeSoBlock) ([]*Bl
 	return ancestors, nil
 }
 
-// addBlockToBlockIndex adds the block to the block index and uncommitted blocks map.
+// addBlockToBlockIndex adds the block to the block index with the given status. It also writes the block to the block
+// index in badger.
```

Comment:
it'll be cleaner to define wrapper functions for addBlockToBlockIndex that set the block's status and internally so that we can reduce the bitwise operations externally

```suggestion
func (bc *Blockchain) storeBlockInBlockIndex(block *MsgDeSoBlock) {
...
}

func (bc *Blockchain) storeValidatedBlockInBlockIndex(block *MsgDeSoBlock) {
...
}

func (bc *Blockchain) sotredCommitedBlockInBlockIndex(block *MsgDeSoBlock) {
...
}

// addBlockToBlockIndex adds the block to the block index with the given status. It also writes the block to the block
// index in badger.
```

Reply:
I actually had the same thought as I was reviewing #755. With wrappers, it's easier to reason about different stages of block status, and it's easier to trace through these status changes.